app/src/main/java/com/garethevans/church/opensongtablet/StageMode.java

```diff
@@ -7467,44 +7566,138 @@ protected void onPostExecute(String s) {
             }
         }
     }
+
+    // IV - code supporting intentional page turns when using pedal for next/previous.
+    // IV - 'Are you sure?' is displayed and the user can repeat the action to continue after 2 seconds
+    // IV - After continue there is a 10s grace period where further pedal use is not tested.  Any pedal 'page' or 'scroll' use extends a further 10s grace period.
+    private void pedalPreviousAndNextConfirm() {
+        if (FullscreenActivity.isSong && (!(StaticVariables.whichMode != null && StaticVariables.whichMode.equals("Stage") &&
+                StaticVariables.songSections != null && StaticVariables.currentSection > 0 && StaticVariables.currentSection < StaticVariables.songSections.length - 2))) {
+            if (StaticVariables.pedalPreviousAndNextNeedsConfirm) {
+                StaticVariables.myToastMessage = getString(R.string.pedal) + " - " + getString(R.string.areyousure);
+                ShowToast.showToast(StageMode.this);
+                StaticVariables.pedalPreviousAndNextNeedsConfirm = false;
+                pedalPreviousAndNextNeedsConfirmHandler.postDelayed(pedalPreviousAndNextNeedsConfirmRunnable, 10000);
+                StaticVariables.pedalPreviousAndNextIgnore = true;
+                // Use a runnable to end the ignore period.
+                pedalPreviousAndNextIgnoreHandler.postDelayed(pedalPreviousAndNextIgnoreRunnable, 2000);
+            } else {
+                try {
+                    // If pedal used again in the ignore period - extend the ignore period.
+                    pedalPreviousAndNextIgnoreHandler.removeCallbacks(pedalPreviousAndNextIgnoreRunnable);
```

Comment:
Not seen how or why this is implemented yet

Reply:
You will :-) It is in pedal press logic.  You will need a pedal to see it in action.  It perhaps needs to be an option that can be used.  It addresses the panic cause by a wrong foot press changing songs during performance.

Reply:
Still can't see this happening (tried it with a pedal) - I'm never shown the 'Are you sure' dialog.
Once we figure this out, I think that it should be a selectable option.  Most users would probably not want that alert showing when first running the app, well until they get their first issue of false presses on stage (my keyboard player does that to me every so often!)

Reply:
Must be pedal.  Do not use the pedal for 10+ seconds (like when performing a song).  Do a previous or next pedal action.  No go?  Strange.

Selectable yes - just a wrap with a preference and more menu stuff!

I was also not certain on the 2s delay. But if you are pedal press happy it need the intentional break.

Reply:
Ok I've figured this out - tracing back through your code (something I realise takes time!), this only happens when performing songs in a set.  This might be useful for non set work as well, e.g. if you go off the planned set and move to another song.
I'll amend my IanTest branch and after pulling your changes to Master, I'll pull request my IanTest to master as well.
I'll add a toggle setting too

Reply:
Ahh.  I had forgotten.  Yes it is currently for set use only.  I can see your point that it might be useful elsewhere but (I think) I thought it was best not to intrude too much.   Perhaps an option to protect all song changes or just when in a set. Solved.

A quirk that I noticed is that you can wander into a set song whilst moving around the song list and the set logic then takes over!  So the next move is within the set.

Reply:
Believe it or not, that was intentional!  Often I deviate from a set (setView is then off).  Then I pull up a song from the song menu that is back in the set, then the app puts me back into set mode.
I did this because when not moving through a set, navigating back to a set item always took me ages to find the where the song was.  It's quicker to find it in the song menu and have the fact that it is labelled as a set item throw me back to the correct point of the set.
The fix to add the are you sure prompt was pretty straightforward, so I've added.
Once I merge your pull request to master and push my additions, I'll get you to update your fork and test the changes.

Just the metronome issue to fix and we're there!

Reply:
I think this might be the the most subtly complex bit of code in the commit - with runnables being fired and reset to get a reasonable effect.  It is aimed at gatekeeping the first song move but then allowing the user to move around freely - as they might then want to quickly move around in a set using the pedal.  It becomes active again 10s after they stop moving.  So not every page move but hopefully the most 'dangerous' one!   The intent is as a 'stage' performance protection feature.

Reply:
As a further comment - the absence of previous and next buttons when not in a set could be considered.  Maybe always have buttons but somehow indicate use in a set (a different icon?).  A thought.

Reply:
![image](https://user-images.githubusercontent.com/68599243/88420542-ec5a5c00-cdde-11ea-8a21-acd236b64d62.png)
A suggestion for an icon!

Reply:
![image](https://user-images.githubusercontent.com/68599243/88420760-49eea880-cddf-11ea-85d9-96a9110398db.png)